### PR TITLE
propagate actor events and Receipt::events_root.

### PR DESCRIPTION
**Pull request by raulk**
Part of https://github.com/filecoin-project/ref-fvm/issues/1048.

---

This is the FFI side implementation of https://github.com/filecoin-project/ref-fvm/pull/1049.
FIP at https://github.com/filecoin-project/FIPs/pull/483.

## TODO

Prior to merging:

- [x] Remove patches: merge and release https://github.com/filecoin-project/ref-fvm/pull/1049, and update here.

### CODE REVIEW

File: rust/Cargo.toml [hunk 1]
```diff
@@ -69,3 +69,10 @@ opencl = ["filecoin-proofs-api/opencl", "bellperson/opencl", "storage-proofs-por
 cuda = ["filecoin-proofs-api/cuda", "bellperson/cuda", "storage-proofs-porep/cuda", "rust-gpu-tools/cuda", "fvm3/cuda", "fvm2/cuda"]
 multicore-sdr = ["storage-proofs-porep/multicore-sdr"]
 c-headers = ["safer-ffi/headers"]
+
+
+[patch.crates-io]
```
Comment (vyzo) [inline rust/Cargo.toml:74]: release for these?

**Review by vyzo — APPROVED** [review]
LGTM modulo cargo patches.

File: rust/Cargo.toml (hunk 1 above)
Comment (raulk) [reply rust/Cargo.toml:74]: Yes, once approved and released.

File: rust/src/fvm/machine.rs [hunk 2]
```diff
@@ -280,6 +281,19 @@ fn fvm_machine_execute_message(
...
         };
 
+        let events = if apply_ret.events.is_empty() {
+            None
+        } else {
+            // This field is informational, if for whatever reason we fail to serialize
+            // return a None and move on. What's important for consensus is the
+            // events_root in the receipt.
+            to_vec(&apply_ret.events)
+                .ok()
```
Comment (Stebalien) [inline rust/src/fvm/machine.rs:291]: We should at least log something _very loudly_.

**Review by Stebalien — APPROVED** [review]

**Comment by arajasek** [comment]
Hey, I don't think this should have gone into `master`. Doing so means we can't update FFI in Lotus until the changes in https://github.com/filecoin-project/lotus/tree/feat/nv18-events land in Lotus `master`, which isn't a place I want to be (especially around an upgrade). In general, I think Lotus  master needs to be compatible with FFI master.

Can we revert and land in a staging branch?

**Pull request merged**